CONTRIBUTING.md

```diff
@@ -47,3 +47,73 @@ mvn -f sdk/{projectForlderDir}/pom.xml -Dgpg.skip clean install
+version_data.txt - Contains Data library and versions
+update_versions.py - This is just a basic python script that will climb through the source tree and update POM and README files. The script utilizies tags within the files to do replacements and the tags are slightly different between the POM and README files.
+set_versions.py - This script should only be used by the build system when we start producing nightly ops builds. 
+
+In POM files this is done by inserting a specifically formatted comment on the same line as the version element.
+```xml
+  <groupId>MyGroup</groupId>
+  <artifactId>MyArtifact</artifactId>
+  <version>1.0.0-preview.1</version> <!-- {x-version-update;MyGroup:MyArtifact;[current|dependency]} -->
+```
+The last element of the tag would be current or dependency depending on the criteria previously explained.
+
+In README files this ends up being slightly different. Because the version tag is inside an XML element that we've explicitly telling a user to copy/paste into their product the comment tag really didn't make sense here. Instead there are tags before and after the XML element tags which effectively says "there's a version somewhere in between these two tags, when you find the line that matches replace it with the appropriate version of the group:artifact defined in the tag."
+
+\[//]: # ({x-version-update-start;MyGroup:MyArtifact;dependency})
+\```xml
+  <groupId>MyGroup</groupId>
+  <artifactId>MyArtifact</artifactId>
+  <version>1.0.0-preview.1</version>
+\```
+\[//]: # ({x-version-update-end})
+
+
+What if I've got something that, for whatever reason, shoudln't be updated? There's a tag for that
+<!-- {x-version-exempt;<groupId>:<artifactId>;reason for excemption} -->
+In theory, absence of an x-version-update tag would do the same thing but the tooling is still being developed and eventually there will be checkin blockers if xml has a version element with no tag.
+
+
+### What does the process looks like?
+Let's say we've GA'd and I need to tick up the version of azure-storage libraries how would I do it? 
+1. I'd open up eng\versioning\version_client.txt and update the current-versions of the libraries are built and released as part of the azure storage pipeline. This list can be found in pom.service.xml under the sdk/storage directory. It's worth noting that any module entry starting with "../" isn't something that's released as part of the pipeline and once we GA these build dependencies for library components outside of a given area should go away and be replaced with downloading the appropriate dependency from Maven like we do for exxternal dependencies.
+2. Execute the update_versions python script
+python eng/versioning/update_versions.py --ut libary --bt client
+This will go through the entire source tree and update all of the references in the POM and README files with the updated versions. Git status will show all of the modified files.
+3. Review and submit a PR with the modified files.
+
```

Comment:
It might also be worth adding a note that we want to introduce automation to automatically bump the versions after a release.